tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/UsernamePasswordIntegrationTests.cs

```diff
@@ -336,6 +357,82 @@ private void AssertTelemetryHeaders(HttpSnifferClientFactory factory, bool IsFai
                 Assert.AreEqual(InvalidGrantError, httpTelemetryRecorder.ErrorCode.FirstOrDefault());
             }
         }
+
+        #region Azure AD Kerberos Feature Tests
+        [TestMethod]
+        public async Task Kerberos_ROPC_AAD_Async()
+        {
+            var labResponse = await LabUserHelper.GetDefaultUserAsync().ConfigureAwait(false);
+            await KerberosRunHappyPathTestAsync(labResponse).ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        public async Task Kerberos_ROPC_ADFSv4Federated_Async()
+        {
+            var labResponse = await LabUserHelper.GetAdfsUserAsync(FederationProvider.AdfsV4, true).ConfigureAwait(false);
+            await KerberosRunHappyPathTestAsync(labResponse).ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        public async Task Kerberos_ROPC_ADFSv4Federated_WithMetadata_Async()
```

Comment:
I don't think this special case where we pass in the federation metadata is interesting for Kerberos. This is just a covenience feature for organizations that haven't configured federation metadata well and it allows app developers to inject their own metadata instead of MSAL discovering it. Nothing to do with Kerberos I think, I would delete this test.

Reply:
Removed.